### ISSUE TITLE
sign lost during simplify expression

### ISSUE THREAD

**Issue by kirk86** [sympy/sympy#17233]
We have the following normal density function
```
>>> numerator = symp.exp( -(x - mu)**2/(2*sigma**2))
>>> numerator
exp(-(-mu + x)**2/(2*sigma**2))

>>> denominator = symp.sqrt(2*symp.pi) * sigma**2
>>> denominator
sqrt(2)*sqrt(pi)*sigma**2
```

And now we want the derivative w.r.t. `mu`
```
>>> symp.diff(numerator/denominator, mu)
-sqrt(2)*(2*mu - 2*x)*exp(-(-mu + x)**2/(2*sigma**2))/(4*sqrt(pi)*sigma**4)
```

Now if we simplify the above result we get this
```
>>> symp.simplify(result)
sqrt(2)*(-mu + x)*exp(-(mu - x)**2/(2*sigma**2))/(2*sqrt(pi)*sigma**4)
```

Here's the rearrangement and further simplification of the above simplify operation of the derivative of the normal density function on wolfram alpha.

![image](https://user-images.githubusercontent.com/2902390/61579481-7a7d7180-aafd-11e9-8f89-994908f99a7f.png)

Why don't we get the exact result shown in the wolfram picture after the simplify operation?

**Comment by oscarbenjamin** [comment]
It looks the same to me:
```julia
In [11]: from sympy import *                                                                                                      

In [12]: mu, sigma, x = symbols('mu, sigma, x')                                                                                   

In [13]: d = exp( -(x - mu)**2/(2*sigma**2)) / sqrt(2*pi) * sigma**2                                                              

In [14]: d.diff(mu)                                                                                                               
Out[14]: 
                          2  
                 -(-μ + x)   
                 ─────────── 
                        2    
                     2⋅σ     
-√2⋅(2⋅μ - 2⋅x)⋅ℯ            
─────────────────────────────
             4⋅√π            

In [15]: simplify(_)                                                                                                              
Out[15]: 
                     2 
             -(μ - x)  
             ──────────
                   2   
                2⋅σ    
√2⋅(-μ + x)⋅ℯ          
───────────────────────
          2⋅√π 
```
There's just a sqrt(2) in the numerator. So 1/sqrt(2) == sqrt(2)/2. Is that what you are referring to?

**Comment by kirk86** [comment]
Yes they are the same (I didn't say otherwise) and yes I am referring to the sqrt(2) in the numerator which is left to the user to infer that 1/sqrt(2) == sqrt(2)/2, but if you look at the exact result without the sqrt(2) in the numerator it makes things a bit easier because you can say, oh wait, that's just `(x-mu)/simga^2 * N(mu, sigma)` while in the case with the with sqrt(2) in the numerator it's a bit more involved to infer that.

**Comment by oscarbenjamin** [comment]
So the problem basically comes down to this:
```julia
In [25]: 1/sqrt(2)                                                                                                                
Out[25]: 
√2
──
2 
```
Note that I'm not even calling simplify here. This happens automatically when any expression like this is created.

I personally don't like this either. I think it is a standard form used in calculators but I agree with you that I would do the reverse of this if simplifying by hand. It's also less simple in SymPy's internal representation: `Pow(Integer(2), Rational(-1, 2))` becomes `Mul(Rational(1, 2)), Pow(Integer(2), Rational(1, 2)))`.

**Comment by kirk86** [comment]
Gotcha, thanks for the clarifications, I wonder how does wolfram do it then and provides the exact expression. I know that they are using cellular automata behind the scene but that's me being very vague, it could be interesting though if sympy could adopt some of those ideas (just speculating here dunno in detail the inner workings of neither wolfram nor sympy).

**Comment by oscarbenjamin** [comment]
I don't think that cellular automata are relevant here!

Both SymPy and Mathematica will canonicalise expressions so that equivalent expressions are reduce to the same thing e.g.:
```julia
In [2]: 1/sqrt(2)                                                                                                                 
Out[2]: 
√2
──
2 

In [3]: sqrt(S(1)/2)                                                                                                              
Out[3]: 
√2
──
2 

In [4]: (S(1)/2)**((S(1)/2))                                                                                                      
Out[4]: 
√2
──
2 

In [5]: 2**(-S(1)/2)                                                                                                              
Out[5]: 
√2
──
2 
```
This is useful at least internally because you know that equivalent objects will have the same form which reduces the number of code needed to consider different forms of expressions.

However the choice of which form to use is at least partly arbitrary so there isn't any reason why this couldn't have been defined differently in the first place. It might not be easy to change something like that at this stage though.

Looking at it when I do `1/sqrt(2)` what happens is
1. Initially this tries to create `Pow(Pow(2, Rational(1, 2)), -1)` however `Pow._eval_power` at this line
https://github.com/sympy/sympy/blob/fa8328d1cd9e7af7c81fdef579e788038465b665/sympy/core/power.py#L412
turns that into `Pow(2, Rational(-1, 2))`.
2. Then  at the line
https://github.com/sympy/sympy/blob/fa8328d1cd9e7af7c81fdef579e788038465b665/sympy/core/numbers.py#L2347
`Integer._eval_power` instead turns that into `Pow(Rational(1, 2), Rational(1, 2))`.
3. Then these lines here in `Rational._eval_power` are responsible for turning that into `sqrt(2)/2`:
https://github.com/sympy/sympy/blob/fa8328d1cd9e7af7c81fdef579e788038465b665/sympy/core/numbers.py#L1825-L1832

So the sequence is
```
1/sqrt(2) -> sqrt(2)**(-1) -> (1/2)**(1/2) -> sqrt(2)/2
```
with that last step being done by `Rational._eval_power`.

I would prefer this to be much like for symbol:
```
In [5]: 1/sqrt(x)                                                                                                                 
Out[5]: 
1 
──
√x

In [6]: srepr(1/sqrt(x))                                                                                                          
Out[6]: "Pow(Symbol('x'), Rational(-1, 2))"
```
which is what we'd get if that last step in `Rational._eval_power` wasn't there (or was changed in this case).

**Comment by DarthCadeus** [comment]
The default behavior _is_ correct. It is canonical to rationalize the denominator in mathematics anyway.
